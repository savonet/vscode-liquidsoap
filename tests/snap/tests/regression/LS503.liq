# In LS-503 we realized that a source may throw an
# exception during output_get_ready call in the initial
# main phase. This code reproduces the issue by throwing
# an exception in output.icecast.

# Reopen stderr to /dev/null to
# disable printing expected exception
#reopen.stderr("/dev/null")

on_cleanup(test.pass)

def on_error(_) =
  shutdown()
  3.
end

p = noise()
o = output.icecast(%wav,fallible=true,host="nonexistent",
                    mount="test",on_error=on_error,p)
