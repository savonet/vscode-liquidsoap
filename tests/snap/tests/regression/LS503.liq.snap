># In LS-503 we realized that a source may throw an
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># exception during output_get_ready call in the initial
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># main phase. This code reproduces the issue by throwing
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># an exception in output.icecast.
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
># Reopen stderr to /dev/null to
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
># disable printing expected exception
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>#reopen.stderr("/dev/null")
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap comment.line.number-sign.liquidsoap
>
>on_cleanup(test.pass)
#^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>
>def on_error(_) =
#^^^ source.liquidsoap keyword.control.liquidsoap
#   ^^^^^^^^^^^^^^^ source.liquidsoap
>  shutdown()
#^^^^^^^^^^^^^ source.liquidsoap
>  3.
#^^ source.liquidsoap
#  ^ source.liquidsoap constant.numeric.integer
#   ^^ source.liquidsoap
>end
#^^^ source.liquidsoap keyword.control.liquidsoap
>
>p = noise()
#^^^^^^^^^^^^ source.liquidsoap
>o = output.icecast(%wav,fallible=true,host="nonexistent",
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                                           ^ source.liquidsoap string.quoted.double.liquidsoap
#                                            ^^^^^^^^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                                                       ^ source.liquidsoap string.quoted.double.liquidsoap
#                                                        ^^ source.liquidsoap
>                    mount="test",on_error=on_error,p)
#^^^^^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
#                          ^ source.liquidsoap string.quoted.double.liquidsoap
#                           ^^^^ source.liquidsoap string.quoted.double.liquidsoap
#                               ^ source.liquidsoap string.quoted.double.liquidsoap
#                                ^^^^^^^^^^^^^^^^^^^^^^ source.liquidsoap
>